openedx/core/djangoapps/catalog/management/commands/cache_programs.py

```diff
@@ -48,33 +143,27 @@ def handle(self, *args, **options):
         except:  # pylint: disable=bare-except
             logger.error('Failed to retrieve program UUIDs.')
             raise
-
-        total = len(uuids)
-        logger.info('Received {total} UUIDs.'.format(total=total))
-
-        programs = {}
+        logger.info('Received {total} UUIDs for site {sitename}'.format(
+            total=len(uuids),
+            sitename=site.domain
+        ))
+        logger.info(str(uuids))
+        return uuids
+
+    def fetch_program_details(self, client, site, uuids, programs):
         failure = False
         for uuid in uuids:
             try:
-                logger.info('Requesting details for program {uuid}.'.format(uuid=uuid))
-                program = client.programs(uuid).get(exclude_utm=1)
-
                 cache_key = PROGRAM_CACHE_KEY_TPL.format(uuid=uuid)
-                programs[cache_key] = program
+                if cache_key not in programs:
+                    logger.info('Requesting details for program {uuid}.'.format(uuid=uuid))
+                    program = client.programs(uuid).get(exclude_utm=1)
+                    programs[cache_key] = program
             except:  # pylint: disable=bare-except
-                logger.exception('Failed to retrieve details for program {uuid}.'.format(uuid=uuid))
+                logger.exception('Failed to retrieve details for program {uuid} for site {site_name}.'.format(
```

Comment:
No need to include site details in this message. The UUID is enough to identify the missing program.